.github/workflows/build-and-test.yml

```diff
@@ -395,7 +395,7 @@ jobs:
       - name: Push Docker Image
         if: steps.check.outputs.passed == 'true'
         run: |
-            docker login -u $DOCKER_USERNAME -p $DOCKER_PASSWORD
+            echo "${{ secrets.DOCKER_PASSWORD }}" | docker login -u ${{ secrets.DOCKER_USERNAME }} --password-stdin
```

Comment:
Perhaps we could try the login action, like we use from the operator?

https://github.com/open-telemetry/opentelemetry-operator/blob/72ba88cb530458feb7aeed3e30bbd75e56d181e5/.github/workflows/publish-images.yaml#L55-L61